packages/flutter_driver/lib/src/extension/extension.dart

```diff
@@ -546,8 +546,27 @@ class FlutterDriverExtension {
     final GetText getTextCommand = command as GetText;
     final Finder target = await _waitForElement(_createFinder(getTextCommand.finder));
     // TODO(yjbanov): support more ways to read text
-    final Text text = target.evaluate().single.widget as Text;
-    return GetTextResult(text.data);
+    final Widget widget = target.evaluate().single.widget;
+    switch (widget.runtimeType) {
+      case RichText:
+        final RichText richText = widget as RichText;
+        if (richText.text.runtimeType == TextSpan)
+          return GetTextResult((richText.text as TextSpan).text);
+        break;
+      case EditableText:
+        return GetTextResult((widget as EditableText).controller.text);
+      case TextField:
+        return GetTextResult((widget as TextField).controller.text);
+      case TextFormField:
+        return GetTextResult((widget as TextFormField).controller.text);
+      case Text:
+        return GetTextResult((widget as Text).data);
+      default:
+        break;
+    }
```

Comment:
This may be easier to follow without the early returns interleaved with the breaks:
```dart
String text;
final Type widgetRuntimeType = widget.runtimeType;

if (widgetRuntimeType == RichText) {
  final RichText richText = widget as RichText;
  if (richText.text.runtimeType == TextSpan) {
    text = (richText.text as TextSpan).text;
  } 
} else if (widgetRuntimeType == EditableText) {
  text = (widget as EditableText).controller.text;
} else if ...
...

if (text == null) {
  throw UnsupportedError(...);
}
return GetTextResult(text);
```